### PR TITLE
Feature/user profile

### LINKED ISSUE #9: Component | User Profile
- [x] structure, plan and style user profile

### PR DESCRIPTION
**Pull request by JoeRupp**
## What was changed:
Add user profile page.

## Explain the reason for changes:
Now when the user visits the profile page, they will see their profile information, including a banner image, profile pic, name, email, and sections with craft items.

Error handing has also been added, so that if the user is not logged in, they will be directed to the log in page if they visit the profile page. 

Up next is testing for this page, adding crafts to the inventory, and making a component to toggle color theme of the profile. 

## Link to issue:
Closes #9

### CODE REVIEW

File: src/Components/UserProfile.js [hunk 1]
```diff
@@ -1,24 +1,118 @@
...
+        />
+        <SectionHeader>
+          You're not signed in! Click the button below to sign in or create an
+          account.
+        </SectionHeader>
+        <Button name={"Login"} link={"/login"} />
+      </UserProfileSection>
+    );
+  }
+
```
Comment (stephanie-roe) [inline src/Components/UserProfile.js:32]: great error handling!

File: src/Components/UserProfile.js [hunk 2]
```diff
@@ -1,24 +1,118 @@
...
+      <CraftDisplaySection></CraftDisplaySection>
+      <Button name={"Add Craft"} link={"/contribution"} />
+      <SectionHeader style={{ color: `${colors[currentColor]}` }}>
+        Borrowed Supplies
+      </SectionHeader>
+      <CraftDisplaySection></CraftDisplaySection>
+      <Button name={"Log Out"} link={"/login"} />
+    </UserProfileSection>
   );
 };
```
Comment (stephanie-roe) [inline src/Components/UserProfile.js:64]: I'm really liking the structure of the profile page- very engaging and organized!

**Review by stephanie-roe — APPROVED** [review]
Profile page looks great! Once you build out those ancillary components like you mentioned, it'll be super cool

File: src/Components/UserProfile.js [hunk 3]
```diff
@@ -1,24 +1,118 @@
...
+  if (!currentUser) {
+    return (
+      <UserProfileSection className="user-profile">
+        <SignInImage
+          className="not-signed-in-img"
+          // src={require(`../assets/`)}
+          alt="not signed in image"
+        />
+        <SectionHeader>
+          You're not signed in! Click the button below to sign in or create an
```
Comment (Abumpus1) [inline src/Components/UserProfile.js:25]: Awesome that you added some error handling!

**Pull request merged**